DigitalLearningSolutions.Web/Controllers/TrackingSystem/CourseSetup/AdminFieldsController.cs

```diff
@@ -56,29 +60,53 @@ public IActionResult Index(int customisationId)
         [Route("{customisationId:int}/AdminFields/{promptNumber:int}/Edit/Start")]
         public IActionResult EditAdminFieldStart(int customisationId, int promptNumber)
         {
+            var courseAdminField = courseAdminFieldsService.GetCourseAdminFieldsForCourse(
+                    customisationId
+                ).AdminFields
+                .Single(cp => cp.PromptNumber == promptNumber);
+
+            var data = new EditAdminFieldData
+            {
+                PromptNumber = courseAdminField.PromptNumber,
+                Prompt = courseAdminField.PromptText,
+                OptionsString = NewlineSeparatedStringListHelper.JoinNewlineSeparatedList(
+                    courseAdminField.Options
+                ),
+                IncludeAnswersTableCaption = true,
+            };
+
+            multiPageFormService.SetMultiPageFormData(
+                data,
+                MultiPageFormDataFeature.EditAdminField,
+                TempData
+            );
+
             return RedirectToAction("EditAdminField", new { customisationId, promptNumber });
         }
 
         [HttpGet]
         [Route("{customisationId:int}/AdminFields/{promptNumber:int}/Edit")]
         [ServiceFilter(typeof(VerifyAdminUserCanManageCourse))]
+        [ServiceFilter(typeof(RedirectMissingMultiPageFormData<EditAdminFieldData>))]
         public IActionResult EditAdminField(int customisationId, int promptNumber)
         {
-            var courseAdminField = courseAdminFieldsService.GetCourseAdminFieldsForCourse(
-                    customisationId
-                ).AdminFields
-                .Single(cp => cp.PromptNumber == promptNumber);
-
-            var data = TempData.Get<EditAdminFieldData>();
+            var data = multiPageFormService.GetMultiPageFormData<EditAdminFieldData>(
+                MultiPageFormDataFeature.EditAdminField,
+                TempData
+            );
 
-            var model = data?.EditModel ?? new EditAdminFieldViewModel(courseAdminField);
+            if (promptNumber != data.PromptNumber)
+            {
+                return RedirectToAction("EditAdminFieldStart", new { promptNumber, customisationId });
+            }

```

Comment:
We don't really need the prompt number apart from for the EditAdminFieldStart action, but it is in the route for this stuff. If we remove it we get a warning on the Route that the parameter is not passed in.
If we don't do the redirect, then our temp data will be seen as valid and then the Edit that is done will apply to the original promptNumber since we go through the rest of the journey only using the model property that is passed backwards and forwards.

Reply:
Removed the prompt number/field number from the edit journeys and moved it in the Start route for each so that a mid part of the route doesn't disappear.